xbmc/pvr/guilib/GUIEPGGridContainerModel.cpp

```diff
@@ -629,17 +663,20 @@ int CGUIEPGGridContainerModel::GetLastEventBlock(const std::shared_ptr<CPVREpgIn
 bool CGUIEPGGridContainerModel::IsEventMemberOfBlock(const std::shared_ptr<CPVREpgInfoTag>& event,
                                                      int iBlock) const
 {
-  const int iFirstBlock = GetBlock(event->StartAsUTC());
-  if (iFirstBlock == iBlock)
+  const int iFirstBlock = GetFirstEventBlock(event);
+  const int iLastBlock = GetLastEventBlock(event);
+
+  if (iFirstBlock > iLastBlock)
+  {
+    return false;
+  }
+  else if (iFirstBlock == iBlock)
```

Comment:
No real need for the braces, but up to you.

Reply:
I think those redundant braces increase readability, thus would kike to keep them.